src/main/scala/cromwell/binding/values/WdlFileLike.scala

```diff
@@ -0,0 +1,10 @@
+package cromwell.binding.values
+
+import cromwell.binding.types.WdlFileType
+
+/**
+ * WDL value which identifies the location of a File-like object.
+ */
+trait WdlFileLike extends WdlPrimitive {
+  val wdlType = WdlFileType
```

Comment:
I suppose my reasoning was more like there should be `WdlFileLikeType` from which `WdlFileType` and `WdlUriType` inherit.  From WDL code, nothing would change, users would still only use `File` or `Uri` as data types.

I'm not sure if that level of abstraction is really necessary though